ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -284,7 +284,7 @@ public final int doPhase2Abort ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
-		if ((s == org.omg.CosTransactions.Status.StatusCommitted)
+		if ((s == org.omg.CosTransactions.Status.StatusCommitted && preparedList.size() == 0)
```

Comment:
Okay so it about completing/implementing what was originally intended in which case fine.
And my question about changing the state from committed to aborting - so that is OK because the [ActionStatus] state model in internal to our implementation and provided our test suite is not throwing up any issues we are fine to introduce new state transitions?